viewer/packages/3d-overlays/src/overlay3DIcon.frag

```diff
@@ -10,10 +10,14 @@ out vec4 fragmentColor;
 
 void main() {
   vec4 textureSample = texture(map, gl_PointCoord); 
+  vec3 computedColor = mix(vColor, vec3(1.0), textureSample.r) ;
+  vec2 centeredPointCoord = gl_PointCoord*2.0 - 1.0;
   
-  if (textureSample.a <= 0.1) {
-    discard;
-  }
-  // TODO 2023-05-05 Savokr: Revisit blending algorithm.
-  fragmentColor = vec4(vColor * textureSample.rgb * colorTint, textureSample.a * collectionOpacity);
+  #if defined(circular_overlay)
+    if (length(centeredPointCoord) > 1.0) {
```

Comment:
more efficient to do `dot(centeredPointCoord, centeredPointCoord) > 1.0`

Reply:
Removed in new implementation.